x-pack/plugins/elastic_assistant/package.json

```diff
@@ -5,6 +5,9 @@
   "private": true,
   "license": "Elastic License 2.0",
   "scripts": {
-    "evaluate-model": "node ./scripts/model_evaluator"
+    "evaluate-model": "node ./scripts/model_evaluator",
+    "openapi:generate": "node scripts/openapi/generate",
+    "openapi:generate:debug": "node --inspect-brk scripts/openapi/generate",
+    "openapi:bundle": "node scripts/openapi/bundle"
```

Comment:
@YulNaumenko, just a heads up that I went ahead and added local scripts & CI integration for the OpenAPI generator within the `elastic_assistant` plugin. I did a cursory review of https://github.com/elastic/kibana/pull/173487/ beforehand to make sure I wasn't duplicating anything (and to try to keep conflicts to a minimum for you), but let me know if I'm missing anything here. 

I also added a bundle script too, as detailed in https://github.com/elastic/kibana/pull/171526, to create a single OAS file to make things easier on the docs side. All this should just 'work' with your branch, but let me know if you see any issues.

Reply:
The `yarn openapi:generate` command may fail when run from any directory (e.g. `x-pack/plugins/elastic_assistant`) with with an error like

```
🪄   Generating new artifacts
Unhandled Promise rejection detected:

[Error: EACCES: permission denied, open '/Users/firstname.gen.ts'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '/Users/firstname.gen.ts'
}
```

if `whoami` returns something like `firstname.lastname`

The workaround for this issue is to (locally) change the implementation of the `getGeneratedFilePath` function in  `packages/kbn-openapi-generator/src/lib/get_generated_file_path.ts`

from

```ts
export function getGeneratedFilePath(sourcePath: string) {
  return sourcePath.replace(/\..+$/, '.gen.ts');
}
```

to

```ts
export function getGeneratedFilePath(sourcePath: string) {
  return sourcePath.replace('.schema.yaml', '.gen.ts');
}
```

and then re-run `yarn openapi:generate`.

Reply:
Draft PR with fix: https://github.com/elastic/kibana/pull/174718